### PR TITLE
Temperature Refactor and Fixes

### LINKED ISSUE #33210: Simple mobs NEVER warm up after freezing
1) Be Gorilla
2) Be Exposed to space
3) Continue taking exposure damage inside comfy rooms, check body temperature, its -15C it never goes up for the rest of the round

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
fix: Fixed frost oil (cryosting) sending insulated crewmembers below absolute zero temperatures.
fix: Fixed damaging cold temps being a death sentence to simple mobs (they wouldn't heat back up to safe temps at room temperature).
refactor: Refactored human temperature code, insulating clothing now helps you stay warm but going unprotected will help you cool off. However, you will now benefit from colder temps when you're overheating even in full EVA/hardsuit gear and you will warm up faster at room temperature in insulated gear as well (i.e. putting on a winter coat will no longer suppress your temperature if your already cold).
/:cl:

Someone should double check my math but currently mob temps are a circus. Aside from the two bugs I mentioned you've got silly behavior like freezing to death LONGER if you wear cold protection or getting no benefit from colder temperatures if your overheating in a cold-proof outfit. Just like high temps overwhelm heat protection, low temps will allow excess heat to bleed off, albeit slower than you would just from being naked. Also we had caps for gaining/losing temp but the "losing" cap was a positive number in a min function, meaning it effectively didn't exist. This made it easy to recover from high temps while in an assistant loadout but insulated crew like the HOS (65% reduction in cooling) or anyone in a space suit (100% reduction in cooling) was virtually doomed to cook themselves into crit. 
  
Fixes #33210

### CODE REVIEW

**Comment by Robustin** [comment]
Also we somehow had no define for the default bodytemp so feel free to yell at me where you see ~310 degrees stuffed in our temp code.

**Comment by QualityVan** [comment]
obligatory spacewalk/perma escape balance memes

**Comment by ShizCalev** [comment]
#33210

**Comment by Robustin** [comment]
This shouldn't affect how quickly you get freezing  or how quickly you overheat, it just makes recovering from this temps a little more rational.

File: code/__DEFINES/atmospherics.dm
Comment (ShizCalev) [inline code/__DEFINES/atmospherics.dm]: 310.15

Comment (Robustin) [reply code/__DEFINES/atmospherics.dm]: But why

Comment (ShizCalev) [reply code/__DEFINES/atmospherics.dm]: Because that's the accurate temperature average of the human body when converting it into kelvin.

Comment (MoreRobustThanYou) [reply code/__DEFINES/atmospherics.dm]: There is less than 0.3 F difference between the 310K and 310.15K

Comment (ShizCalev) [reply code/__DEFINES/atmospherics.dm]: And there is a 1.8F difference between 311 and 310.

Comment (MoreRobustThanYou) [reply code/__DEFINES/atmospherics.dm]: also @Robustin is using weebeditor

Comment (ike709) [reply code/__DEFINES/atmospherics.dm]: Just use 67.8 Felsius

**Comment by Robustin** [comment]
The hell is going on with Travis

**Comment by ShizCalev** [comment]
It's been fucky the last few weeks.

**Comment by kevinz000** [comment]
please do a testmerge of this before merging...

**Comment by kevinz000** [comment]
also if this ruins tempgun meta i'm calling you robustindedplsnerf

**Comment by Cyberboss** [comment]
Good label

File: code/__DEFINES/atmospherics.dm
Comment (ninjanomnom) [reply code/__DEFINES/atmospherics.dm]: 310.15, what @ShizCalev said

**Review by ninjanomnom — DISMISSED** [review]

**Comment by epochayur** [comment]
wow, an actually good robustin PR

File: code/modules/mob/living/carbon/human/species.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species.dm]: elevate `thermal_protection = 1` to above L1461 and have it set within ifs (or within the proc) rather than having 2 separated vars.

File: code/modules/mob/living/carbon/human/species.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species.dm]: The equations are the same in both Hot & Cold. Perhaps turn the equation into a singular proc with arguments defining which set of defines to use.

**Comment by Robustin** [comment]
IIRC cold recovery is actually nerfed for low insulation loadouts but buffed for higher insulation. Unless the assistant runs into an actually hot room, tempgun got a tiny buff v. greytide.

File: code/__DEFINES/atmospherics.dm
Comment (ShizCalev) [inline code/__DEFINES/atmospherics.dm]: 360.15

File: code/__DEFINES/atmospherics.dm
Comment (ShizCalev) [inline code/__DEFINES/atmospherics.dm]: 260.15

**Review by ShizCalev — DISMISSED** [review]

File: code/__DEFINES/atmospherics.dm
Comment (ShizCalev) [reply code/__DEFINES/atmospherics.dm]: or just make it (BODYTEMP_NORMAL + 50)

File: code/__DEFINES/atmospherics.dm
Comment (ShizCalev) [reply code/__DEFINES/atmospherics.dm]: or just make it (BODYTEMP_NORMAL - 50)

**Comment by Robustin** [comment]
It takes a special level of ass to make a new PR adding a conflicting define just because you're obsessed with having the body temp at 98.6F instead of 98.3F - nevermind the part where our bodies can be 180 degrees in this game before we take heat damage.

Also update the tags.

**Comment by ExcessiveUseOfCobblestone** [comment]
what did he mean by this...

**Comment by Robustin** [comment]
@ShizCalev whined about wanting the temp at 310.15K instead of 310. I said the whole reason we had inconsistencies in our temperature code was because of people forgetting to put .15 after every temp change involving mobs and so from a "maintainer" perspective it made no sense to keep pushing that approach just for the sake of whatever realism argument there is for having mob's default temp being 98.6F over 98.3F when humans can run around with 180F or 30F body temps without taking damage - realism seems moot.

So the "maintainer" instead opened a new PR that introduced the same define that I added but set it to 310.15, forcing me to change this PR to use it.

It was a few minutes of extra work but I hate the idea of maintainers making petty little moves like that to increase the work of people making fixes just to win a pointless argument and make temp code less maintainable in the long run.

**Comment by ExcessiveUseOfCobblestone** [comment]
That's why you use defines :^)

**Comment by ShizCalev** [comment]
7 out of 71 lines changed in my PR were conflicting. Figured it was easier to just correct the places that didn't use defines and were inconsistent from the start (that you weren't hitting anyway) to use a define, then just resolve the minor conflict, than it was to spam you with 67 different change request links.

**Comment by ShizCalev** [comment]
Need the WIP tag removed too?

**Comment by optimumtact** [comment]
if you just listend to them in the first place instead of being a salty crybaby, you wouldn't be in this situation

**Review by AnturK — CHANGES_REQUESTED** [review]
Quick local test : go in space get TOO HOT message, then after admin heal the too hot alert stays on.

**Comment by Robustin** [comment]
I fixed that and another issue during local testing but it only causes two more problems that I have no apparent cause. I'm rapidly losing hope that I'm going to be able to iron this thing out.

**Comment by Robustin** [comment]
Ok that should be all the fixes, going to have to do another round of testing

**Comment by Robustin** [comment]
soooo Travis...

**Pull request merged**
